Panic when host is not defined

When a host is not defined for and endpoint, the middleware will throw a panic since it's assuming all endpoints will have a host backend defined (and it should).

Config example:
```
{
  "version": 2,
  "extra_config": {},
  "timeout": "3000ms",
  "cache_ttl": "300s",
  "output_encoding": "json",
  "name": "testing",
  "endpoints": [
    {
      "endpoint": "/new2",
      "method": "GET",
      "extra_config": {},
      "output_encoding": "json",
      "concurrent_calls": 1,
      "backend": [
        {
          "url_pattern": "/v1/employees",
          "encoding": "json",
          "sd": "static",
          "extra_config": {}
        }
      ]
    }
  ]
}
```
Error:
```
[GIN-debug] GET    /new1                     --> github.com/devopsfaith/krakend/router/gin.CustomErrorEndpointHandler.func1 (3 handlers)
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/devopsfaith/krakend-amqp.backendFactory.initConsumer(0x24c9d80, 0xc0000aa040, 0x24d7080, 0xc000331a40, 0xc0001adbf0, 0xc0004a9530, 0xc000570790, 0x24c9d80, 0xc0000aa040, 0xc0003e1300, ...)
	/go/pkg/mod/github.com/devopsfaith/krakend-amqp@v0.0.0-20190315181526-5f14dc177694/consumer.go:28 +0x159d
github.com/devopsfaith/krakend-amqp.backendFactory.New(0x24c9d80, 0xc0000aa040, 0x24d7080, 0xc000331a40, 0xc0001adbf0, 0xc0004a9530, 0xc000570790, 0xc0003e1300, 0x100)
	/go/pkg/mod/github.com/devopsfaith/krakend-amqp@v0.0.0-20190315181526-5f14dc177694/client.go:35 +0x7a
```

--- a/consumer.go
+++ b/consumer.go
@@ -17,6 +17,7 @@ import (
 const consumerNamespace = "github.com/devopsfaith/krakend-amqp/consume"
 
 var errNoConsumerCfgDefined = errors.New("no amqp consumer defined")
+var errNoBackendHostDefined = errors.New("no host backend defined")
 
 type consumerCfg struct {
 	queueCfg
@@ -25,6 +26,10 @@ type consumerCfg struct {
 }
 
 func (f backendFactory) initConsumer(ctx context.Context, remote *config.Backend) (proxy.Proxy, error) {
+	if len(remote.Host) < 1 {
+		return proxy.NoopProxy, errNoBackendHostDefined
+	}
+
 	dns := remote.Host[0]
 
 	cfg, err := getConsumerConfig(remote)

